back/src/Guide/Database/Set.hs

```diff
@@ -0,0 +1,584 @@
+        SET enabled_sections = $2
+        WHERE uid = $1
+        |]
+      encoder = contrazip2 uidParam itemSectionSetParam
+      decoder = HD.noResult
+  lift $ HT.statement (catId, newSections) (Statement sql encoder decoder False)
+
+-- | Add itemId to category items_order.
+addItemIdToCategory :: Uid Category -> Uid Item -> ExceptT DatabaseError Transaction ()
+addItemIdToCategory catId itemId = do
+    itemsOrder <- getCategoryItemsOrder catId
+    let addItemId =
+          if notElem itemId itemsOrder then itemId : itemsOrder else itemsOrder
+    let sql = [r|
+          UPDATE categories
+          SET items_order = $2
+          WHERE uid = $1
+          |]
+        encoder = contrazip2 uidParam uidsParam
+        decoder = HD.noResult
+    lift $ HT.statement
+      (catId, addItemId) (Statement sql encoder decoder False)
+
+-- | Delete itemId from category items_order.
+deleteItemIdFromCategory :: Uid Category -> Uid Item -> ExceptT DatabaseError Transaction ()
+deleteItemIdFromCategory catId itemId = do
+    itemsOrder <- getCategoryItemsOrder catId
+    let deleteItemId =
+          if elem itemId itemsOrder then filter (/= itemId) itemsOrder else itemsOrder
+    let sql = [r|
+          UPDATE categories
+          SET items_order = $2
+          WHERE uid = $1
+          |]
+        encoder = contrazip2 uidParam uidsParam
+        decoder = HD.noResult
+    lift $ HT.statement
+      (catId, deleteItemId) (Statement sql encoder decoder False)
+
+----------------------------------------------------------------------------
+-- Items
+----------------------------------------------------------------------------
+
+-- | Set item name.
+setItemName :: Uid Item -> "name" :! Text -> ExceptT DatabaseError Transaction ()
+setItemName itemId (arg #name -> name) = do
+  let sql = [r|
+        UPDATE items
+        SET name = $2
+        WHERE uid = $1
+        |]
+      encoder = contrazip2 uidParam textParam
+      decoder = HD.noResult
+  lift $ HT.statement (itemId, name) (Statement sql encoder decoder False)
+
+-- | Set item link.
+setItemLink :: Uid Item -> "link" :? Url -> ExceptT DatabaseError Transaction ()
+setItemLink itemId (argF #link -> link) = do
+  let sql = [r|
+        UPDATE items
+        SET link = $2
+        WHERE uid = $1
+        |]
+      encoder = contrazip2 uidParam textParamNullable
+      decoder = HD.noResult
+  lift $ HT.statement (itemId, link) (Statement sql encoder decoder False)
+
+-- | Set item hackage.
+--
+-- "hackage" :? Text == Maybe Text with label.
+setItemHackage :: Uid Item -> "hackage" :? Text -> ExceptT DatabaseError Transaction ()
+setItemHackage itemId (argF #hackage -> hackage) = do
+  let sql = [r|
+        UPDATE items
+        SET hackage = $2
+        WHERE uid = $1
+        |]
+      encoder = contrazip2 uidParam textParamNullable
+      decoder = HD.noResult
+  lift $ HT.statement (itemId, hackage) (Statement sql encoder decoder False)
+
+-- | Set item summary.
```

Comment:
TODO: add a note that this function should not be used without doing a `get` first in the same transaction. I will add the note later.